### PR TITLE
make all tests inherit from BaseTestClass

### PR DESCRIPTION
**Pull request by frank-dong-ms**
RT

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -20,52 +20,3 @@ dotnet_diagnostic.MSML_NoBestFriendInternal.severity = none
 
 # MSML_NoInstanceInitializers: No initializers on instance fields or properties
 dotnet_diagnostic.MSML_NoInstanceInitializers.severity = none
-
-[test/Microsoft.ML.CodeAnalyzer.Tests/**.cs]
-# BaseTestClass does not apply for analyzer testing.
-# MSML_ExtendBaseTestClass: Test classes should be derived from BaseTestClass
-dotnet_diagnostic.MSML_ExtendBaseTestClass.severity = none
```
Comment (sharwell) [inline .editorconfig]: 📝 This one shouldn't change

File: test/Microsoft.ML.CodeAnalyzer.Tests/Code/BaseTestClassTest.cs [hunk 2]
```diff
@@ -5,18 +5,24 @@
...
 using VerifyCS = Microsoft.ML.CodeAnalyzer.Tests.Helpers.CSharpCodeFixVerifier<
-    Microsoft.ML.InternalCodeAnalyzer.BaseTestClassAnalyzer,
-    Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
+Microsoft.ML.InternalCodeAnalyzer.BaseTestClassAnalyzer,
+Microsoft.CodeAnalysis.Testing.EmptyCodeFixProvider>;
 
 namespace Microsoft.ML.CodeAnalyzer.Tests.Code
 {
-    public class BaseTestClassTest
+    public class BaseTestClassTest : BaseTestClass
```
Comment (sharwell) [inline test/Microsoft.ML.CodeAnalyzer.Tests/Code/BaseTestClassTest.cs]: 📝 This project shouldn't change as part of the rest of the work.

File: .editorconfig (hunk 1 above)
Comment (frank-dong-ms) [reply .editorconfig]: Yes, you are right, thanks

---
In reply to: [381542429](https://github.com/dotnet/machinelearning/pull/4858#discussion_r381542429) [](ancestors = 381542429)

File: test/Microsoft.Extensions.ML.Tests/Microsoft.Extensions.ML.Tests.csproj [hunk 3]
```diff
@@ -10,10 +10,13 @@
     <ProjectReference Include="..\..\src\Microsoft.ML.Transforms\Microsoft.ML.Transforms.csproj" />
     <ProjectReference Include="..\..\src\Microsoft.ML.StandardTrainers\Microsoft.ML.StandardTrainers.csproj" />
     <ProjectReference Include="..\Microsoft.ML.TestFrameworkCommon\Microsoft.ML.TestFrameworkCommon.csproj" />
+    <ProjectReference Include="..\Microsoft.ML.TestFramework\Microsoft.ML.TestFramework.csproj" />
```
Comment (yaeldekel) [inline test/Microsoft.Extensions.ML.Tests/Microsoft.Extensions.ML.Tests.csproj]: >TestFramework [](start = 47, length = 13)

There was a change recently to move some functionality out of Microsoft.ML.TestFramework and into a separate project Microsoft.ML.TestFrameworkCommon. I don't remember what was the reason for this change (I think it had something to do with the nightly build test project).
Anyway, if all test classes need to inherit from `BaseTestClass`, shouldn't we move that class to Microsoft.ML.TestFrameworkCommon?

Comment (frank-dong-ms) [reply test/Microsoft.Extensions.ML.Tests/Microsoft.Extensions.ML.Tests.csproj]: BaseTestClass has reference to this Contracts which is internal class of Microsoft.ML.Core and referenced by TestFramework by bestfriend. 
I don't know why this Contracts class is in src folder not test folder but that can be a different issue.

---
In reply to: [381876287](https://github.com/dotnet/machinelearning/pull/4858#discussion_r381876287) [](ancestors = 381876287)

**Pull request merged**